src/common.c

```diff
@@ -1374,23 +1378,27 @@ ssize_t ofi_bsock_recvv(struct ofi_bsock *bsock, struct iovec *iov, size_t cnt)
 		ofi_byteq_add(&bsock->rq, (size_t) ret);
 		assert(ofi_bsock_readable(bsock));
 		bytes += ofi_byteq_readv(&bsock->rq, iov, cnt, bytes);
-		return bytes;
+		*len = bytes;
+		return 0;
 	}
 
 	/* It's too difficult to adjust the iov without copying it, so return
 	 * what data we have.  The caller will consume the iov and retry.
 	 */
-	if (bytes)
-		return bytes;
+	if (bytes) {
+		*len = bytes;
+		return 0;
+	}
 
 	ret = bsock->sockapi->recvv(bsock->sockapi, bsock->sock, iov, cnt,
 				    MSG_NOSIGNAL, &bsock->rx_sockctx);
-	if (ret > 0)
-		return ret;
+	if (ret > 0) {
+		*len = ret;
+		return 0;
+	}
 out:
 	assert(ret != -OFI_EINPROGRESS_URING);
-	if (bytes)
-		return bytes;
+	*len = bytes;
```

Comment:
This change here and the same in the above function are the problem.  bytes can be > 0, so *len is set correctly, but now we fall through to the return call below, which can return a negative errno instead of 0.  We can take this path if we read data from the bsock buffer (bytes > 0), but still have more data to read.  That can call recv() at line 1372 to refill the bsock buffer, which can return EAGAIN.

Reply:
Thank you very much for your help! I fixed the bug and resubmitted the first 2 patches. If they pass, I will resubmit the whole patch series.

Reply:
I did some tests over the week-end and my conclusions are as follows:
- AppVeyor is always failing the jobs `Debug-EFA`, which is unrelated to my changes. Actually, all recent PRs fail the same tests.
- I tried a first version where I added the following return statement at the end of the function, right after the `out:` label (in both `ofi_bsock_recv` functions):
```
if (ret == -FI_EAGAIN)
    return 0;
```
With above changes, Intel and AWS both fail. 
- I updated the PR with the following change, which is more in line with the code before my changes:
```
if (*len)
    return 0;
```
AWS seems happy now, but Intel CI keeps failing. I leave the PR in this state in case you want to look at the failure in Intel CI.